src/lib/hasher/solana.ts

```diff
@@ -0,0 +1,415 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { createHash } from 'crypto'
+import { PublicKey } from '@solana/web3.js'
+import BN from 'bn.js'
+import { type Schema, serialize } from 'borsh'
+import type { CCIPMessage, CCIPVersion } from '../types.ts'
+import type { LeafHasher } from './common.ts'
+
+// Message types from https://github.com/smartcontractkit/chainlink-ccip/blob/main/chains/solana/contracts/programs/ccip-offramp/src/messages.rs
+/**
+ * RampMessageHeader
+ * - message_id: fixed 32 bytes array.
+ * - source_chain_selector, dest_chain_selector, sequence_number, nonce: u64 values.
+ */
+class RampMessageHeader {
+  message_id: Uint8Array
```

Comment:
Why Uint8Array? We use strings everywhere in this codebase, and Atlas implementation is also siding with keeping bytes32 0x-encoded messageIds (since they are 32 bytes on every supported chain)